dist/tools/jlink: remove jlink prompt

jasonatran opened the pull request:
When flashing over JTAG, the prompt "J-Link> " appears, and the user needs to press enter to continue with a flash. Segger removed this at one point, but they seem to have brought it back for legacy users...

kaspar030 commented:
Would this lead to problems with older JLinkExe versions?

jasonatran commented:
It's possible older JLinkExe versions do not have this option. Unfortunately, Segger does not publish the older versions on their site to test with. One of the Segger forum moderators stated the behavior will stay the same for future versions:

http://forum.segger.com/index.php?page=Thread&threadID=2998

basilfx commented:
Haven't upgraded JLink yet, but isn't a similar fix needed for the `do_reset`, `do_debug` and `do_debugserver`? Or is it only required for flashing?

A-Paul commented:
@jasonatran you're welcome.

As it is, the script is an adapter to the RIOT build system and its mostly controlled by environment variables. The JLink is also capable of "speaking" SWD, and hardwiring JTAG operation may block that option. Therefor I woudn't recommend change the script that way.

~~You can always feed additional command line options into the script.~~
At a fist glance I found an issue with processing of commandline parameters.

I have to dive deeper into the topic.

@kaspar030 Do you know of any advantage using an older version?

@basilfx That's a valid point. The parameter affects every communication between JLink and the device. If it makes sense at all, it has to be changed for all commands.

kYc0o commented:
Let's discuss it for the next release.

A-Paul commented:
> Let's discuss it for the next release.

@kYc0o, fine with me.
After a quick talk with @haukepetersen it is obvious that the script needs some additional care regarding the processing of command line parameters.

kYc0o commented:
Ok let's see afterwards. BTW @jasonatran this one needs the commit name to be changed too.

jasonatran commented:
How deep can you go with the directory name in the commit name?

I'll look into the scripts a little more after the next release to see what ideas I can come up with.

kYc0o commented:
> How deep can you go with the directory name in the commit name?

As long as it doesn't exceed 80 characters :)

miri64 commented:
Postponed due to feature freeze

OlegHahm commented:
@jasonatran, are you still willing to update this PR and get it merged for the release? Otherwise, I would propose to close it.

jasonatran commented:
I'm willing to update this PR. I'll get to it when I can. @A-Paul, could you provide more feedback on this change?

A-Paul commented:
@jasonatran can you please add your fix to every call to `${JLINK}`? Can you also check if the binary behind `${JLINK_SERVER}` also needs a additional parameter?

kYc0o commented:
What's missing on this PR?

A-Paul commented:
@kYc0o, the changes I asked for in https://github.com/RIOT-OS/RIOT/pull/5620#issuecomment-273333704

jasonatran commented:
Rebased and updated. @A-Paul, the binary behind ${JLINK_SERVER} (i.e. JLinkGDBServer) does not seem to need the additional parameter. The only new change is the additional parameter to do_reset() in jlink.sh.

kYc0o commented:
Well, Murdock agrees but Jenkins don't. I tried to restart the job but I don't know if I succeeded...

OlegHahm commented:
Jenkins' approval is still optional.

kYc0o commented:
Well if @A-Paul ACKs then it's ok.
